build/dockerfiles/Dockerfile

```diff
@@ -0,0 +1,71 @@
+FROM debian:bullseye@sha256:71cb300d5448af821aedfe63afd55ba05f45a6a79f00dcd131b96b780bb99fe4
+
+# Set system time to 00:00:01 UTC
+ENV SOURCE_DATE_EPOCH=1
+
+# Avoid prompts during installation
+ENV DEBIAN_FRONTEND=noninteractive
+
+# Install curl
+RUN apt-get update && apt-get install -y curl jq
+
+# Copy apt install files from repo
+COPY apt/packages/ ./
+
+# Copy digest for apt install files
+COPY apt/package_urls.txt ./
+
+# File containing the list of packages and hashes
+ENV PACKAGE_LIST_FILE="./package_urls.txt"
+
+# Download install files, check hashes, and install
+RUN while IFS= read -r line; do \
+        package_url=$(echo "$line" | awk '{print $1}'); \
+        package_hash=$(echo "$line" | awk '{print $2}'); \
+        package_name=$(basename "$package_url"); \
+        if [ ! -f "./$package_name" ]; then \
+            echo "Downloading $package_url"; \
+            curl -O "$package_url"; \
+        fi; \
+        echo "Verifying hash for $package_name"; \
+        echo "$package_hash *$package_name" > temp_checksum_file; \
+        sha256sum -c temp_checksum_file; \
+        if [ $? -ne 0 ]; then \
+            echo "Hash check failed for $package_name"; \
+            exit 1; \
+        fi; \
+    done < "$PACKAGE_LIST_FILE" \
+    && dpkg -i ./*.deb
```

Comment:
It should only install all the files from the list above, otherwise we might get unexpected packages.

Reply:
This part of code is no longer present!